src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php

```diff
@@ -1184,11 +1184,22 @@ private function registerTranslatorConfiguration(array $config, ContainerBuilder
                 $files[$locale][] = (string) $file;
             }
 
+            $projectDirLength = \strlen($projectDir = $container->getParameter('kernel.project_dir'));
+
             $options = array_merge(
                 $translator->getArgument(4),
                 [
                     'resource_files' => $files,
-                    'scanned_directories' => array_merge($dirs, $nonExistingDirs),
+                    'scanned_directories' => $scannedDirectories = array_merge($dirs, $nonExistingDirs),
+                    'cache_vary' => [
+                        'scanned_directories' => array_map(static function (string $dir) use ($projectDir, $projectDirLength): string {
+                            if (0 === strncmp($dir, $projectDir, $projectDirLength)) {
+                                return ltrim(substr($dir, $projectDirLength), '/');
+                            }
+
+                            return $dir;
```

Comment:
looks like this would be enough and a bit stricter:
`return 0 === strpos($dir, $projectDir.'/') ? substr($dir, 1 + \strlen($projectDir) : $dir;`